File: src/LanAudioRelay.Core/Models/ReceiverAnnouncement.cs
namespace LanAudioRelay.Core.Models;

public sealed record ReceiverAnnouncement(
    string Id,
    string Name,
    string HostName,
    string IpAddress,
    int ControlPort,
    int ProtocolVersion)
{
    public string DisplayName => $"{Name} ({IpAddress})";
}

File: mac/LanAudioRelayMac/Sources/LanAudioRelayMacCore/ReceiverAnnouncement.swift
import Foundation

public struct ReceiverAnnouncement: Identifiable, Hashable, Sendable {
    public let id: String
    public let name: String
    public let hostName: String
    public let ipAddress: String
    public let controlPort: Int
    public let protocolVersion: Int

    public var displayName: String {
        "\(name) (\(ipAddress))"
    }
}
